src/lib/pipeline/ReorderBuffer.cc

```diff
@@ -34,39 +35,31 @@ void ReorderBuffer::reserve(const std::shared_ptr<Instruction>& insn) {
 }
 
 void ReorderBuffer::commitMicroOps(uint64_t insnId) {
-  if (buffer_.size()) {
-    size_t index = 0;
-    int firstOp = -1;
-    bool validForCommit = false;
-
-    // Find first instance of uop belonging to macro-op instruction
-    for (; index < buffer_.size(); index++) {
-      if (buffer_[index]->getInstructionId() == insnId) {
-        firstOp = index;
-        break;
-      }
-    }
+  size_t bsize = buffer_.size();
+  if (!bsize) return;
 
-    if (firstOp > -1) {
-      // If found, see if all uops are committable
-      for (; index < buffer_.size(); index++) {
-        if (buffer_[index]->getInstructionId() != insnId) break;
-        if (!buffer_[index]->isWaitingCommit()) {
-          return;
-        } else if (buffer_[index]->isLastMicroOp()) {
-          // all microOps must be in ROB for the commit to be valid
-          validForCommit = true;
-        }
-      }
-      if (!validForCommit) return;
+  size_t index = 0;
+  uint8_t ins_cnt = 0;
```

Comment:
To avoid confusion akin to the other comment left on this variable, perhaps rename this to `uopCount` or `microOpCount`?